integration/update_test.go

```diff
@@ -466,6 +466,58 @@ func TestUpdateSet(t *testing.T) {
 	}
 }
 
+func TestUpdateTimestamp(t *testing.T) {
+	t.Parallel()
+
+	// store the current timestamp with $currentDate operator;
+	t.Run("currentDateReadBack", func(t *testing.T) {
+		maxDifference := time.Duration(2 * time.Second)
+		now := time.Now()
+		nowTimestamp := primitive.Timestamp{T: uint32(now.Unix()), I: uint32(0)}
+		id := "string-empty"
+
+		update := bson.D{{"$currentDate", bson.D{{"value", bson.D{{"$type", "timestamp"}}}}}}
+		stat := &mongo.UpdateResult{
+			MatchedCount:  1,
+			ModifiedCount: 1,
+			UpsertedCount: 0,
+		}
+		path := types.NewPathFromString("value")
+		result := bson.D{{"_id", id}, {"value", nowTimestamp}}
+
+		ctx, collection := setup(t, shareddata.Scalars, shareddata.Composites)
+
+		res, err := collection.UpdateOne(ctx, bson.D{{"_id", id}}, update)
+		require.NoError(t, err)
+		require.Equal(t, stat, res)
+
+		// read it, check that it is close to the current time;
+		var actualB bson.D
+		err = collection.FindOne(ctx, bson.D{{"_id", id}}).Decode(&actualB)
+		require.NoError(t, err)
+
+		expected := ConvertDocument(t, result)
+		actualX := ConvertDocument(t, actualB)
+
+		testutil.CompareAndSetByPathTime(t, expected, actualX, maxDifference, path)
+
+		// write a new timestamp value with the same time;
+		updateB := bson.D{{"$set", bson.D{{"value", nowTimestamp}}}}
+		expectedB := bson.D{{"_id", id}, {"value", nowTimestamp}}
```

Comment:
`expectedB` is equal to `result` defined above.

Reply:
> `expectedB` is equal to `result` defined above.

It will be confusing if I'd be comparing to the variable named as `result` and threaten it as expected.